cpp/oneapi/dal/algo/knn/backend/gpu/infer_kernel_brute_force_dpc.cpp

```diff
@@ -421,7 +425,16 @@ static infer_result<Task> kernel(const context_gpu& ctx,
         }
     }
 
-    if (is_euclidean_distance) {
+    if (is_cosine_distance) {
+        using dst_t = pr::cosine_distance<Float>;
+        [[maybe_unused]] constexpr auto order = get_ndorder(train_data);
+        using search_t = pr::search_engine<Float, dst_t, order>;
+
+        const dst_t dist{ queue };
+        const search_t search{ queue, train_data, train_block, dist };
+        search(query_data, callback, infer_block, neighbor_count).wait_and_throw();
+    }
```

Comment:
Minor; maybe distances better to check in this order: euclidean, minkowski, chebyshev, cosine.

Reply:
It seems like there is no difference)

Reply:
There is a difference if you want to point at the most used branch. As far as Euclidean distance is widely used and default for kNN it makes sense to place it first. But you are right, there are no difference in results, that's why I marked it as minor =)